src/Omega_h_modify.cpp

```diff
@@ -369,13 +370,14 @@ static void assign_new_numbering(Read<T> old_ents2new_numbers,
       rep_self_count = (rep2md_order[mod_dim].exists() ? 1 : 0);
     }
     auto mods2prods_dim = mods2prods[mod_dim];
+    auto mods2mds_dim = mods2mds[mod_dim];
     if (rep2md_order[mod_dim].exists()) {
       /* this is the upward adjacency case */
       OMEGA_H_CHECK(mods2mds[mod_dim].exists());
-      auto mods2mds_dim = mods2mds[mod_dim];
       auto rep2md_order_dim = rep2md_order[mod_dim];
       auto write_prod_offsets = OMEGA_H_LAMBDA(LO mod) {
         auto md = mods2mds_dim[mod];
+        OMEGA_H_CHECK(rep2md_order_dim[md] >= 0);
```

Comment:
This check might be slightly faster if the value is loaded into a local variable first.